hll/src/main/java/io/druid/hll/HyperLogLogCollector.java

```diff
@@ -284,6 +284,11 @@ protected ByteBuffer getStorageBuffer()
     return storageBuffer;
   }
 
+  public ByteBuffer createStorageBufferCopy()
```

Comment:
I guess this is needed because `getStorageBuffer()` is protected. The method name is kind of misleading; we're not actually copying the buffer, just creating another pointer to it.

IMO, it'd be nicer to instead add a method like `public static HyperLogLogCollector makeCollectorSharingStorage(HyperLogLogCollector otherCollector)` which uses a duplicate of otherCollector's storageBuffer.